src/evo/deterministicmns.cpp

```diff
@@ -470,11 +470,17 @@ void CDeterministicMNList::AddMN(const CDeterministicMNCPtr& dmn, bool fBumpTota
         throw(std::runtime_error(strprintf("%s: Can't add a masternode %s with a duplicate collateralOutpoint=%s", __func__,
                 dmn->proTxHash.ToString(), dmn->collateralOutpoint.ToStringShort())));
     }
-    for (const CService& entry : dmn->pdmnState->netInfo.GetEntries()) {
-        if (!AddUniqueProperty(*dmn, entry)) {
-            mnUniquePropertyMap = mnUniquePropertyMapSaved;
-            throw(std::runtime_error(strprintf("%s: Can't add a masternode %s with a duplicate address=%s", __func__,
-                                               dmn->proTxHash.ToString(), entry.ToStringAddrPort())));
+    for (const NetInfoEntry& entry : dmn->pdmnState->netInfo.GetEntries()) {
+        if (const auto& service_opt{entry.GetAddrPort()}; service_opt.has_value()) {
+            const CService& service{service_opt.value()};
+            if (!AddUniqueProperty(*dmn, service)) {
+                mnUniquePropertyMap = mnUniquePropertyMapSaved;
+                throw std::runtime_error(strprintf("%s: Can't add a masternode %s with a duplicate address=%s",
+                                                   __func__, dmn->proTxHash.ToString(), service.ToStringAddrPort()));
+            }
+        } else {
+            throw std::runtime_error(
+                strprintf("%s: Can't add a masternode %s with invalid address", __func__, dmn->proTxHash.ToString()));
```

Comment:
0f3519db961909e3d1b10b83381a17c1a5bd9c03: https://github.com/dashpay/dash/pull/6665#discussion_r2083805250, same for other new "invalid address" errors

Reply:
Since we need to register the underlying type in the unique properties map, this check has to be later expanded for `DomainPort` (see below, WIP code).

https://github.com/dashpay/dash/blob/4f716a5cb9098684a3240f4d66e2a4a2a3c289eb/src/evo/deterministicmns.cpp#L474-L491

As for potentially invalid addresses, we try to prevent construction of an malformed `NetInfoEntry` (see below).

https://github.com/dashpay/dash/blob/66ee55c6eabaa88dc8aaaafcc7fceed3865d9814/src/evo/netinfo.h#L68-L73

But they can still be empty (i.e. invalid) and well-formed so I would still keep this check even if we have earlier checks against this (`IsTriviallyValid()`)

Reply:
IMO, it should be validated much earlier; `AddMN` should return failure if and only if _duplicated_ masternode (with same protx / address /etc are created); all other types of valiure (such as NetInfoEntry members) should be validated earlier.

Either, this code should be just removed (if it's already validated) or some extra validations for NetInfoEntry should be added. Maybe even just throw exception if you try to create NetInfoEntry with invalid object; but `AddMN` it's a bit too late to validate if MN is valid at all